gettsim/interface.py

```diff
@@ -117,6 +122,43 @@ def compute_taxes_and_transfers(
     return results
 
 
+def _fail_if_datatype_is_false(data, columns_overriding_functions, functions):
+    """Check if the provided data has the right types.
+
+    Parameters
+    ----------
+    data : pandas.Series or pandas.DataFrame or dict of pandas.Series
+        Data provided by the user.
+    columns_overriding_functions : str list of str
+        Names of columns in the data which are preferred over function defined in the
+        tax and transfer system.
+    functions : dict of callable
+        A dictionary of functions.
+
+    Returns
+    -------
+    ValueError
+        Fail if the data types are not matching the required in gettsim.
+
+    """
+    check_data = True
+    for column_name, series in data.items():
+        if column_name in STANDARD_DATA_TYPES:
+            internal_type = STANDARD_DATA_TYPES[column_name]
+            check_data = check_if_series_has_internal_type(series, internal_type)
+        elif column_name in columns_overriding_functions:
+            internal_type = functions[column_name].__annotations__["return"]
```

Comment:
We should take care of elements in `columns_overriding_functions` that don't correspond to a function: either ignore those or throw an explicit error (I suggest the latter)

Reply:
There is already `_fail_if_columns_overriding_functions_are_not_in_functions`. It was just not called. Will change!